internal/eval/store_eval.go

```diff
@@ -448,3 +449,154 @@ func evalZRANGE(args []string, store *dstore.Store) *EvalResponse {
+		}
+	}
+
+	key := args[0]
+	obj := store.Get(key)
+
+	// If key doesn't exist prior initial cardinality changes hence return 1
+	if obj == nil {
+		hll := hyperloglog.New()
+		for _, arg := range args[1:] {
+			hll.Insert([]byte(arg))
+		}
+
+		obj = store.NewObj(hll, -1, object.ObjTypeString, object.ObjEncodingRaw)
+
+		store.Put(key, obj)
+		return &EvalResponse{
+			Result: int64(1),
+			Error:  nil,
+		}
+	}
+
+	existingHll, ok := obj.Value.(*hyperloglog.Sketch)
+	if !ok {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrGeneral(diceerrors.WrongTypeHllErr),
+		}
+	}
+	initialCardinality := existingHll.Estimate()
+	for _, arg := range args[1:] {
+		existingHll.Insert([]byte(arg))
+	}
+
+	if newCardinality := existingHll.Estimate(); initialCardinality != newCardinality {
+		return &EvalResponse{
+			Result: int64(1),
+			Error:  nil,
+		}
+	}
+
+	return &EvalResponse{
+		Result: int64(0),
+		Error:  nil,
+	}
+}
+
+func evalPFCOUNT(args []string, store *dstore.Store) *EvalResponse {
+	if len(args) < 1 {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongArgumentCount("PFCOUNT"),
+		}
+	}
+
+	unionHll := hyperloglog.New()
+
+	for _, arg := range args {
+		obj := store.Get(arg)
+		if obj != nil {
+			currKeyHll, ok := obj.Value.(*hyperloglog.Sketch)
+			if !ok {
+				return &EvalResponse{
+					Result: nil,
+					Error:  diceerrors.ErrGeneral(diceerrors.WrongTypeHllErr),
+				}
+			}
+			err := unionHll.Merge(currKeyHll)
+			if err != nil {
+				return &EvalResponse{
+					Result: nil,
+					Error:  diceerrors.ErrGeneral(diceerrors.InvalidHllErr),
+				}
+			}
+		}
+	}
+
+	return &EvalResponse{
+		Result: unionHll.Estimate(),
+		Error:  nil,
+	}
+}
+
+func evalPFMERGE(args []string, store *dstore.Store) *EvalResponse {
+	if len(args) < 1 {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongArgumentCount("PFMERGE"),
+		}
+	}
+
+	var mergedHll *hyperloglog.Sketch
+	destKey := args[0]
+	obj := store.Get(destKey)
+
+	// If destKey doesn't exist, create a new HLL, else fetch the existing
+	if obj == nil {
+		mergedHll = hyperloglog.New()
+	} else {
+		var ok bool
+		mergedHll, ok = obj.Value.(*hyperloglog.Sketch)
+		if !ok {
+			return &EvalResponse{
+				Result: nil,
+				Error:  diceerrors.ErrGeneral(diceerrors.WrongTypeHllErr),
+			}
+		}
+	}
+
+	for _, arg := range args {
+		obj := store.Get(arg)
+		if obj != nil {
+			currKeyHll, ok := obj.Value.(*hyperloglog.Sketch)
+			if !ok {
+				return &EvalResponse{
+					Result: nil,
+					Error:  diceerrors.ErrGeneral(diceerrors.WrongTypeHllErr),
+				}
+			}
+
+			err := mergedHll.Merge(currKeyHll)
+			if err != nil {
+				return &EvalResponse{
+					Result: nil,
+					Error:  diceerrors.ErrGeneral(diceerrors.InvalidHllErr),
+				}
+			}
+		}
+	}
+
+	// Save the mergedHll
+	obj = store.NewObj(mergedHll, -1, object.ObjTypeString, object.ObjEncodingRaw)
+	store.Put(destKey, obj)
+
+	return &EvalResponse{
+		Result: "OK",
```

Comment:
Instead of sending a response as "OK" you can return back `clientio.OK`

Reply:
Ok, I will make this change

Reply:
I have made this change. Also added the Integration Tests for RESP Protocol